### PR TITLE
8323660: Serial: Fix header ordering and indentation

### PR DESCRIPTION
**Pull request by LizBing**
follow up 8234502.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8323660](https://bugs.openjdk.org/browse/JDK-8323660): Serial: Fix header ordering and indentation (**Enhancement** - P4)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [ed54e312](https://git.openjdk.org/jdk/pull/17398/files/ed54e31252d0d4037d3cb93d9578805bcf87e2f6)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [ed54e312](https://git.openjdk.org/jdk/pull/17398/files/ed54e31252d0d4037d3cb93d9578805bcf87e2f6)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17398/head:pull/17398` \
`$ git checkout pull/17398`

Update a local copy of the PR: \
`$ git checkout pull/17398` \
`$ git pull https://git.openjdk.org/jdk.git pull/17398/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17398`

View PR using the GUI difftool: \
`$ git pr show -t 17398`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17398.diff">https://git.openjdk.org/jdk/pull/17398.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17398#issuecomment-1889384507)

### CODE REVIEW

**Comment by albertnetymk** [comment]
Some indentations of arg list are broken, e.g. `SerialHeap::collect_generation`.

File: src/hotspot/share/gc/serial/serialHeap.cpp [hunk 1]
```diff
@@ -44,43 +31,54 @@
...
 #include "runtime/java.hpp"
+#include "runtime/mutexLocker.hpp"
 #include "runtime/threads.hpp"
 #include "runtime/vmThread.hpp"
 #include "services/memoryService.hpp"
 #include "utilities/autoRestore.hpp"
 #include "utilities/debug.hpp"
 #include "utilities/formatBuffer.hpp"
 #include "utilities/macros.hpp"
+#include "services/memoryManager.hpp"
```
Comment (albertnetymk) [inline src/hotspot/share/gc/serial/serialHeap.cpp:81]: This order seems wrong.

**Review by albertnetymk — APPROVED** [review]

**Review by plummercj — APPROVED** [review]
SA change looks good. But...

It looks like all the copyright updates to 2024 are missing from this PR and the previous PR. We may need another PR for that. Also, the two new files in that PR should have a copyright that just says 2024, since that is the year it was added. Instead they say "Copyright (c) 2000, 2023"

**Comment by LizBing** [comment]
> SA change looks good. But...
> 
> It looks like all the copyright updates to 2024 are missing from this PR and the previous PR. We may need another PR for that. Also, the two new files in that PR should have a copyright that just says 2024, since that is the year it was added. Instead they say "Copyright (c) 2000, 2023"

reported.
https://bugs.openjdk.org/browse/JDK-8323693

**Comment by LizBing** [comment]
thanks for the review.
/integrate

File: src/hotspot/share/gc/serial/serialHeap.cpp [hunk 2]
```diff
@@ -92,26 +90,24 @@ SerialHeap* SerialHeap::heap() {
...
+  _rem_set(nullptr),
+  _soft_ref_policy(),
+  _gc_policy_counters(new GCPolicyCounters("Copy:MSC", 2, 2)),
+  _incremental_collection_failed(false),
+  _full_collections_completed(0),
+  _young_manager(nullptr),
+  _old_manager(nullptr),
+  _eden_pool(nullptr),
+  _survivor_pool(nullptr),
+  _old_pool(nullptr) {
```
Comment (kimbarrett) [inline src/hotspot/share/gc/serial/serialHeap.cpp:105]: Don't make this change to the indentation of the mem-initializer-list.  The existing indentation is one of
several styles used in GC code, including throughtout Z.  The proposed new indentation is one which is
also probably still used in older code.  But some time ago there was discussion amongst the Oracle GC
team about constructor formatting and which of several we might prefer.  We didn't reach a consensus
at the time, but this one was unanimously, and strongly, disliked.  The problem is that with the indentation
of the mem-initializer-list being the same as the body, and the opening brace of the body cuddled with
the end of the mem-initializer-list, it is hard to spot where the mem-initializer-list ends and the body
begins.

**Review by kimbarrett — CHANGES_REQUESTED** [review]

**Review by kimbarrett — APPROVED** [review]

**Pull request closed without merging**